src/frontend/src/icp/components/core/IcLoaderWallets.svelte

```diff
@@ -0,0 +1,71 @@
+<script lang="ts">
+	import { onDestroy } from 'svelte';
+	import { initIcrcWalletWorker } from '$icp/services/worker.icrc-wallet.services';
+	import { initIcpWalletWorker } from '$icp/services/worker.icp-wallet.services';
+	import type { WalletWorker } from '$icp/types/ic-listener';
+	import { enabledIcNetworkTokens } from '$lib/derived/network-tokens.derived';
+	import type { TokenId } from '$lib/types/token';
+	import { debounce, nonNullish } from '@dfinity/utils';
+	import { toastsError } from '$lib/stores/toasts.store';
+	import { i18n } from '$lib/stores/i18n.store';
+	import { replacePlaceholders } from '$lib/utils/i18n.utils';
+
+	let workers: Map<TokenId, WalletWorker> = new Map<TokenId, WalletWorker>();
+
+	const manageWorkers = async () => {
+		const unusedWorkers: TokenId[] = Array.from(workers.keys()).filter(
+			(tokenId) => !$enabledIcNetworkTokens.some(({ id: icTokenId }) => icTokenId === tokenId)
+		);
+
+		unusedWorkers.forEach((tokenId) => {
+			const worker = workers.get(tokenId);
+
+			if (nonNullish(worker)) {
+				worker.stop();
+				workers.delete(tokenId);
+			}
+		});
+
+		await Promise.all(
+			$enabledIcNetworkTokens.map(async (token) => {
```

Comment:
for readability maybe extract a function?

```
const loadWorker = async (token) => {...}
await Promise.all($enabledIcNetworkTokens.map(loadWorker))
```

Reply:
done! extracted